source/oauth/webapp.d

```diff
@@ -49,13 +62,9 @@ class OAuthWebapp
         if (!req.session)
             return false;
 
-        if (auto session =
-            settings ? OAuthSession.load(settings, req.session) : null)
+        if (settings !is null)
         {
-            () @trusted {
-                req.context["oauth.session"] = session;
-            } ();
-
+            loadSessionToContext(req, settings);
             return true;
```

Comment:
This would introduce a serious bug. Should only return `true` if a session is available.